### PR TITLE
fix passthrough idle timeout

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fix: https://github.com/istio/istio/issues/38413

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 1]
```diff
@@ -531,6 +533,7 @@ func buildInboundCatchAllFilterChains(configgen *ConfigGeneratorImpl,
 		ConfigType: &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(&tcp.TcpProxy{
 			StatPrefix:       util.BlackHoleCluster,
 			ClusterSpecifier: &tcp.TcpProxy_Cluster{Cluster: util.BlackHoleCluster},
+			IdleTimeout:      durationpb.New(idleTimeoutDuration),
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go:536]: Why do we need this for inbound blackhole cluster?

**Comment by ramaraochavali** [comment]
Looks good except one minor comment. Is it possible to add a test for this?

**Comment by zirain** [comment]
> Looks good except one minor comment. Is it possible to add a test for this?

update UTs.

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM. will defer to @ramaraochavali

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**